### PR TITLE
add RepeatingLoggingThread and startHaltingThread()

### PR DESCRIPTION
**Pull request by jisookim0513**

### CODE REVIEW

File: src/main/scala/com/metamx/common/concurrent/RepeatingLoggingThread.scala [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    }
+  })
+  thread.setName(name)
+
+  def start() = thread.start()
+
+  def cancel() = thread.cancel
+
+  def repeatNow() = thread.repeatNow
+}
```
Comment (xvrl) [inline src/main/scala/com/metamx/common/concurrent/RepeatingLoggingThread.scala]: missing new line

Comment (jisookim0513) [inline src/main/scala/com/metamx/common/concurrent/RepeatingLoggingThread.scala]: fixed

File: src/main/scala/com/metamx/common/concurrent/RepeatingLoggingThread.scala [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        case e: Exception =>
+          log.error(e, "[%s] - Exception while performing operation".format(name))
+      }
+    }
+  })
+  thread.setName(name)
+
+  def start() = thread.start()
+
+  def cancel() = thread.cancel
```
Comment (xvrl) [inline src/main/scala/com/metamx/common/concurrent/RepeatingLoggingThread.scala]: we should probably add () to the underlying cancel method in RepeatingThread

File: src/main/scala/com/metamx/common/concurrent/RepeatingLoggingThread.scala [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+  def repeatNow() = thread.repeatNow
```
Comment (xvrl) [inline src/main/scala/com/metamx/common/concurrent/RepeatingLoggingThread.scala]: same here

**Comment by xvrl** [comment]
LGTM, can you squash your first two commits into one?

File: src/main/scala/com/metamx/common/scala/threads.scala [hunk 4]
```diff
@@ -21,6 +22,19 @@ object threads extends Logging
     initRunnerThread(name, Some(quietPeriod), f)
   }
 
+  def startHaltingThread(body: => Any, name: String) = daemonThread { abortingRunnable {
+    try body catch {
+      case e: Throwable =>
+        log.error(e, "Halting")
+        e.printStackTrace()
```
Comment (xdralex) [inline src/main/scala/com/metamx/common/scala/threads.scala]: why printstacktrace here?

Comment (jisookim0513) [inline src/main/scala/com/metamx/common/scala/threads.scala]: I deleted printStackTrace()

**Pull request merged**